addons/portal/controllers/portal.py

```diff
@@ -246,22 +247,45 @@ def _update_password(self, old, new1, new2):
 
         try:
             request.env['res.users'].change_password(old, new1)
-        except UserError as e:
-            return {'errors': {'password': e.name}}
         except AccessDenied as e:
             # FIXME XMO dead code since https://github.com/odoo/odoo/pull/45723
             # AccessDenied is an UserError now
             msg = e.args[0]
             if msg == AccessDenied().args[0]:
                 msg = _('The old password you provided is incorrect, your password was not changed.')
             return {'errors': {'password': {'old': msg}}}
+        except UserError as e:
+            return {'errors': {'password': e.name}}
 
         # update session token so the user does not get logged out (cache cleared by passwd change)
         new_token = request.env.user._compute_session_token(request.session.sid)
         request.session.session_token = new_token
 
         return {'success': {'password': True}}
 
+    @route('/my/deactivate_account', type='http', auth='user', website=True, methods=['POST'])
+    def deactivate_account(self, validation, password, **post):
+        values = self._prepare_portal_layout_values()
+        values['get_error'] = get_error
+        values['open_deactivate_modal'] = True
+
+        if validation != request.env.user.login:
+            values['errors'] = {'deactivate': 'validation'}
+        else:
+            try:
+                request.env['res.users']._check_credentials(password, {'interactive': True})
+                request.env.user.sudo()._deactivate_portal_user(**post)
+                request.session.logout()
+                return request.redirect('/web/login?message=%s' % urls.url_quote(_('Account de-activated!')))
+            except AccessDenied:
+                values['errors'] = {'deactivate': 'password'}
+            except UserError as e:
+                values['errors'] = {'deactivate': {'other': str(e)}}
+
+        return request.render('portal.portal_my_security', values, headers={
+            'X-Frame-Options': 'DENY',
```

Comment:
the new trend in #89223 appears to be `sameorigin`

Reply:
Are you a youtube creator of trendy videos ?

Reply:
Not yet. Does it involve odoo, beers and fromage fondu ?